scene/gui/rich_text_label.cpp

```diff
@@ -2535,6 +2535,61 @@ Control::CursorShape RichTextLabel::get_cursor_shape(const Point2 &p_pos) const
 	return get_default_cursor_shape();
 }
 
+bool RichTextLabel::_is_mouse_in_text_area(const Vector2 &p_pos) const {
+	Size2 size = get_size();
+#ifdef MATH_CHECKS
+	if (unlikely(size.x < 0 || size.y < 0)) {
+		ERR_PRINT("Rect2 size is negative, this is not supported. Use Rect2.abs() to get a Rect2 with a positive size.");
+	}
+#endif
+
+	if (p_pos.x < 0.0) {
+		return false;
+	}
+	if (p_pos.y < 0.0) {
+		return false;
+	}
+	if (p_pos.x >= size.x) {
+		return false;
+	}
+	if (p_pos.y >= size.y) {
+		return false;
+	}
+	return true;
+}
+
+Vector2 RichTextLabel::_get_projected_mouse_pos(const Vector2 &p_mouse_pos) {
```

Comment:
Why is this called "projected"? You are just clamping it.
btw the method is unnecessary, you can do `p_mouse_pos.clamp(Vectoe2(), get_size())`.